src/components/JobsSearchInput/index.tsx

```diff
@@ -0,0 +1,103 @@
+import { ChangeEvent, MouseEventHandler, useEffect, useRef, useState } from "react";
+import DropdownArrow from "../../../public/images/dropdownArrow.svg";
+
+interface JobsSearchInput {
+  type: 'select' | 'text';
+  label: string;
+  options?: string[];
+  placeholder?: string;
+}
+
+const JobsSearchInput = ({ type, label, options, placeholder }: JobsSearchInput) => {
```

Comment:
Seria importante colocar um I antes do nome pra quando formos declarar algo não chamar a interface e sim o componente.
```suggestion
const JobsSearchInput = ({ type, label, options, placeholder }: IJobsSearchInput) => {
```